src/TestFrameworkOrchestrator.ts

```diff
@@ -0,0 +1,48 @@
+import { TestFrameworkFactory, TestFramework } from 'stryker-api/test_framework';
+import { StrykerOptions } from 'stryker-api/core';
+import * as log4js from 'log4js';
+
+const IGNORE_WARNING = 'Set `coverageAnalysis` option explicitly to "off" to ignore this warning.';
+const log = log4js.getLogger('TestFrameworkOrchestrator');
+
+export default class TestFrameworkOrchestrator {
+
+  constructor(private options: StrykerOptions) {
+  }
+
+  determineTestFramework(): TestFramework {
+    if (this.options.coverageAnalysis !== 'perTest') {
+      log.debug('The `coverageAnalysis` setting is "%s", not hooking into the test framework to achieve performance benefits.', this.options.coverageAnalysis);
+      return null;
```

Comment:
I'll make it `TestFramework | null`. Void, or undefined, represents a deeper absence of a value and should never be used for a functional absence of a value in my opinion. By making it `TestFramework | null` we can (at a later time please) enable [`--strictNullChecks`](https://www.typescriptlang.org/docs/release-notes/typescript-2.0.html) and let the compiler help us to not make mistakes.